### PR TITLE
[conductor] add cleanContext

### PR DESCRIPTION
**Pull request by Yugue**
*Refactored `clearn.dart` to support the frontend calling `RunContext`*

*[Issue](https://github.com/flutter/flutter/issues/91119)*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 7 new digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/93264

File: dev/conductor/core/lib/src/clean.dart [hunk 1]
```diff
@@ -71,6 +70,25 @@ class CleanCommand extends Command<void> {
...
+/// This is a frontend-agnostic implementation.
+class RunContext {
+  RunContext({
+    required this.stateFile,
+  });
+
+  final File stateFile;
+
+  void run() {
     stateFile.deleteSync();
```
Comment (christopherfujino) [inline dev/conductor/core/lib/src/clean.dart:92]: nit: while we're at it, can you make this async (`return stateFile.delete();` and return type `Future<void>` and `await`ed at the call site), that way we don't have to refactor and roll when this involves a network call.

**Comment by christopherfujino** [comment]
> It looks like this pull request may not have tests. Please make sure to add tests before merging. If you need [an exemption](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests) to this rule, contact Hixie on the #hackers channel in [Chat](https://github.com/flutter/flutter/wiki/Chat).
> 
> If you are not sure if you need tests, consider this rule of thumb: the purpose of a test is to make sure someone doesn't accidentally revert the fix. Ask yourself, is there anything in your PR that you feel it is important we not accidentally revert back to how it was before your fix?
> 
> **Reviewers**: Read the [Tree Hygiene page](https://github.com/flutter/flutter/wiki/Tree-hygiene#how-to-review-code) and make sure this patch meets those guidelines before LGTMing.

can you add a unit test?

**Comment by Yugue** [comment]
> > It looks like this pull request may not have tests. Please make sure to add tests before merging. If you need [an exemption](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests) to this rule, contact Hixie on the #hackers channel in [Chat](https://github.com/flutter/flutter/wiki/Chat).
> > If you are not sure if you need tests, consider this rule of thumb: the purpose of a test is to make sure someone doesn't accidentally revert the fix. Ask yourself, is there anything in your PR that you feel it is important we not accidentally revert back to how it was before your fix?
> > **Reviewers**: Read the [Tree Hygiene page](https://github.com/flutter/flutter/wiki/Tree-hygiene#how-to-review-code) and make sure this patch meets those guidelines before LGTMing.
> 
> can you add a unit test?

Would [the existing tests](https://github.com/Yugue/flutter/blob/conductor_cleancontext/dev/conductor/core/test/clean_test.dart) already test this part?

**Comment by skia-gold** [comment]
Gold has detected about 18 new digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/93264

**Comment by christopherfujino** [comment]
> > > It looks like this pull request may not have tests. Please make sure to add tests before merging. If you need [an exemption](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests) to this rule, contact Hixie on the #hackers channel in [Chat](https://github.com/flutter/flutter/wiki/Chat).
> > > If you are not sure if you need tests, consider this rule of thumb: the purpose of a test is to make sure someone doesn't accidentally revert the fix. Ask yourself, is there anything in your PR that you feel it is important we not accidentally revert back to how it was before your fix?
> > > **Reviewers**: Read the [Tree Hygiene page](https://github.com/flutter/flutter/wiki/Tree-hygiene#how-to-review-code) and make sure this patch meets those guidelines before LGTMing.
> > 
> > 
> > can you add a unit test?
> 
> Would [the existing tests](https://github.com/Yugue/flutter/blob/conductor_cleancontext/dev/conductor/core/test/clean_test.dart) already test this part?

True, this is covered by existing tests. However, as part of the framework testing policy, all changes need to have tests, or you need to request an exemption from Hixie. As this is a code change, I think you should write at least one new test.

**Comment by Yugue** [comment]
> > > > It looks like this pull request may not have tests. Please make sure to add tests before merging. If you need [an exemption](https://github.com/flutter/flutter/wiki/Tree-hygiene#tests) to this rule, contact Hixie on the #hackers channel in [Chat](https://github.com/flutter/flutter/wiki/Chat).
> > > > If you are not sure if you need tests, consider this rule of thumb: the purpose of a test is to make sure someone doesn't accidentally revert the fix. Ask yourself, is there anything in your PR that you feel it is important we not accidentally revert back to how it was before your fix?
> > > > **Reviewers**: Read the [Tree Hygiene page](https://github.com/flutter/flutter/wiki/Tree-hygiene#how-to-review-code) and make sure this patch meets those guidelines before LGTMing.
> > > 
> > > 
> > > can you add a unit test?
> > 
> > 
> > Would [the existing tests](https://github.com/Yugue/flutter/blob/conductor_cleancontext/dev/conductor/core/test/clean_test.dart) already test this part?
> 
> True, this is covered by existing tests. However, as part of the framework testing policy, all changes need to have tests, or you need to request an exemption from Hixie. As this is a code change, I think you should write at least one new test.

Done

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**